cpp/src/parquet/arrow/record_reader.cc

```diff
@@ -717,11 +727,96 @@ inline void TypedRecordReader<DType>::ConfigureDictionary(const DictionaryPage*
   DCHECK(current_decoder_);
 }
 
+// If the data page includes repetition and definition levels, we
+// initialize the level decoders and return the number of encoded level bytes.
+// The return value helps determine the number of bytes in the encoded data.
+template <typename DType>
+template <typename PageType>
+int64_t TypedRecordReader<DType>::InitializeLevelDecoders(
+    const std::shared_ptr<PageType> page, const Encoding::type repetition_level_encoding,
+    const Encoding::type definition_level_encoding) {
+  // Read a data page.
+  num_buffered_values_ = page->num_values();
+
+  // Have not decoded any values from the data page yet
+  num_decoded_values_ = 0;
+
+  const uint8_t* buffer = page->data();
```

Comment:
Only `data()` and `num_values()` are used here. We should have a common base class for DataPageV1 and DataPageV2 instead of having this template